src/components/workspace-utils.js

```diff
@@ -340,6 +346,55 @@ export const WorkspaceTagSelect = props => {
   })
 }
 
+export const WorkspaceSubmissionStatusIcon = ({ status, loadingSubmissionStats, size = 20 }) => {
+  return Utils.cond(
+    [loadingSubmissionStats, () => h(DelayedRender, [
+      h(TooltipTrigger, {
+        content: 'Loading submission status',
+        side: 'left'
+      }, [spinner({ size })])
+    ])],
+    [status, () => h(TooltipTrigger, {
+      content: span(['Last submitted workflow status: ', span({ style: { fontWeight: 600 } }, [_.startCase(status)])]),
+      side: 'left'
+    }, [
+      Utils.switchCase(status,
+        ['success', () => icon('success-standard', { size, style: { color: colors.success() } })],
+        ['failure', () => icon('error-standard', { size, style: { color: colors.danger(0.85) } })],
+        ['running', () => icon('sync', { size, style: { color: colors.success() } })]
+      )
+    ])],
+    () => null)
+}
+
+export const RecentlyViewedWorkspaceCard = ({ workspace, submissionStatus, loadingSubmissionStats, timestamp }) => {
+  const { workspace: { namespace, name, googleProject } } = workspace
+
+  const dateViewed = Utils.makeCompleteDate(new Date(parseInt(timestamp)).toString())
+
+  return h(Clickable, {
+    style: { ...Style.elements.card.container, margin: '0 0.25rem', lineHeight: '1.5rem', flex: '0 1 calc(25% - 10px)' },
+    href: Nav.getLink('workspace-dashboard', { namespace, name })
+  }, [
+    div({ style: { flex: 'none' } }, [
+      div({ style: { color: colors.accent(), ...Style.noWrapEllipsis, fontSize: 16, marginBottom: 7 } }, name),
+      div({ style: { display: 'flex', justifyContent: 'space-between' } }, [
+        div({ style: { ...Style.noWrapEllipsis, whiteSpace: 'pre-wrap', fontStyle: 'italic' } }, `Viewed ${dateViewed}`),
+        div({ style: { display: 'flex', alignItems: 'center' } }, [
+          h(WorkspaceSubmissionStatusIcon, {
+            status: submissionStatus,
+            loadingSubmissionStats,
+            size: 16
+          }),
+          !!googleProject ?
+            h(GcpLogo, { title: 'Google Cloud', role: 'img', style: { marginLeft: 5, height: 16 } }) :
```

Comment:
I think we may want more compact icons-- @yunip20 is also looking at this for the ticket she's working on.

Reply:
@yunip20 has some smaller icons in https://github.com/DataBiosphere/terra-ui/pull/3303/files

Reply:
@cahrens @yunip20 Nice! I'll wait for that PR to merge and then rebase this branch to grab those new icons

Reply:
@MatthewBemis There are other things in Yunyi's PR that are going to create merge issues (specifically, the workflows submission icon). I think it would be easier if you go ahead and merge this one, and we will handle the updates/merge conflicts.

Reply:
Works for me- thanks!